app/http.php

```diff
@@ -36,11 +40,57 @@
         // 'document_root' => __DIR__.'/../public',
         // 'enable_static_handler' => true,
         'http_compression' => true,
+        'dispatch_func' => 'dispatch',
         'http_compression_level' => 6,
         'package_max_length' => $payloadSize,
         'buffer_output_size' => $payloadSize,
     ]);
 
+/**
+ * Assigns HTTP requests to worker threads by analyzing its payload/content.
+ *
+ * Routes requests as 'safe' or 'risky' based on specific content patterns (like POST actions or certain domains)
+ * to optimize load distribution between the workers. Utilizes `$safeThreadsPercent` to manage risk by assigning
+ * riskier tasks to a dedicated worker subset. Prefers idle workers, with fallback to random selection if necessary.
+ *
+ * @param Server $server Swoole server instance.
+ * @param string|null $data Request content for categorization.
+ * @global int $workerNumber Total number of workers.
+ * @return int Chosen worker ID for the request.
+ */
+function dispatch(Server $server, $data = null)
+{
+    global $workerNumber;
+
+    $safeThreadsPercent = intval(App::getEnv('_APP_SAFE_THREADS_PERCENT', 80)) / 100;
+
+    $safeThreads = (int) floor($workerNumber * $safeThreadsPercent);
+
+    $risky = false;
+    if ($data && str_contains($data, 'POST') && str_contains($data, '/executions')) {
+        $risky = true;
+    } elseif ($data && str_contains($data, '.appwrite.global')) {
```

Comment:
Can we use the function domain environment variable here instead of hard coding .appwrite.global?

Reply:
what happens if we use custom domain?